sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -954,17 +954,28 @@ CURLcode CurlSession::ReadStatusLineAndHeadersFromRawResponse(
     // /3 containing a "Connection" header should be considered malformed. (HTTP/2:
     // https://httpwg.org/specs/rfc9113.html#ConnectionSpecific
     //  HTTP/3: https://httpwg.org/specs/rfc9114.html#rfc.section.4.2)
-    if (m_response->GetMajorVersion() == 1 && m_response->GetMinorVersion() >= 1)
+    // We assume that the server is not sending malformed responses,
+    // so we are not considering HTTP/2 or HTTP/3 here.
+    if (m_response->GetMajorVersion() == 1 && m_response->GetMinorVersion() == 1)
     {
+      // HTTP/1.1
       m_httpKeepAlive = (!hasConnectionClose || hasConnectionKeepAlive);
     }
-    else if (m_response->GetMajorVersion() <= 1)
+    else if (m_response->GetMajorVersion() == 1 && m_response->GetMinorVersion() == 0)
     {
+      // HTTP/1.0
       m_httpKeepAlive = hasConnectionKeepAlive;
```

Comment:
Note that [HTTP 1.0](https://www.rfc-editor.org/rfc/rfc1945) does *NOT* define a Connection header (I checked the HTTP RFC last night). So there should be no circumstance in which we set m_httpKeepAlive for an HTTP 1.0 response.

If an HTTP 1.0 server response is received with a Connection header, the header should be ignored.

Reply:
I am confused, are we sure that's the behavior we want? I thought in HTTP 1.0, an explicit presence of the keep-alive header is the opt-in for connection persistence.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection#directives
> Indicates that either the client or the server would like to close the connection. This is the default on HTTP/1.0 requests.

To bolster your finding, this does mention the following (citing the HTTP spec):
https://en.wikipedia.org/wiki/HTTP_persistent_connection#HTTP_1.0
> Under HTTP 1.0, connections should always be closed by the server after sending the response.

But also says:
```text
Since at least late 1995, developers of popular products (browsers, web servers, etc.) using HTTP/1.0, started to add an unofficial extension (to the protocol) named "keep-alive" in order to allow the reuse of a connection for multiple requests/responses.
...
When the server receives this request and generates a response, if it supports keep-alive then it also adds the same above header to the response. Following this, the connection is not dropped, but is instead kept open.
```

If what you described Larray is what we want as the behavior, I can certainly make the change to adhere to the spec. Do please confirm/let me know.

Reply:
Discussed offline, we are OK with honoring the presence of the `Connection: keep-alive` header, rather than always setting `m_httpKeepAlive = false` for HTTP/1.0.